### PR TITLE
feat(sast): Add java sast policies

### PR DESCRIPTION
**Pull request by maxamel**
Added 10 Java policies + tests.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/sast/checks/java/InadequateAlgorithmStrength.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+metadata:
+  id: CKV_SAST_29
```
Comment (arielkru) [inline checkov/sast/checks/java/InadequateAlgorithmStrength.yaml:2]:
```suggestion
  id: CKV3_SAST_29
```

File: checkov/sast/checks/java/BrokenCryptographicAlgorithm.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+metadata:
+  id: CKV_SAST_30
```
Comment (arielkru) [inline checkov/sast/checks/java/BrokenCryptographicAlgorithm.yaml:2]:
```suggestion
  id: CKV3_SAST_30
```

File: checkov/sast/checks/java/PersistentCookie.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  - and:
+    - cond_type: "filter"
+      attribute: "pattern"
+      operator: "within"
+      value: |
+        (javax.servlet.http.Cookie $C).setMaxAge($AGE);
+    - cond_type: "variable"
+      variable: "$AGE"
+      operator: "greater_than_or_equal"
+      value: "31536000"
```
Comment (arielkru) [inline checkov/sast/checks/java/PersistentCookie.yaml:21]:
```suggestion
  - cond_type: "filter"
    attribute: "pattern"
    operator: "within"
    value: |
      (javax.servlet.http.Cookie $C).setMaxAge($AGE);
  - cond_type: "variable"
    variable: "$AGE"
    operator: "greater_than_or_equal"
    value: "31536000"
```

File: checkov/sast/checks/java/SensitiveInfoInCookie.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  name: 'Ensure cookie does not contain sensitive session info'
+  guidelines: 'The information stored in a custom cookie should not be sensitive or related to the session.
+    In most cases, sensitive data should only be stored in session and referenced by the users session cookie.'
+  category: 'sast'
+  cwe: "CWE-614: Sensitive Cookie in HTTPS Session Without 'Secure' Attribute"
+scope:
+  languages:
+    - java
+definition:
+  - and:
```
Comment (arielkru) [inline checkov/sast/checks/java/SensitiveInfoInCookie.yaml:12]: redundant

File: checkov/sast/checks/java/TrustBoundary.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    is untrusted. By combining trusted and untrusted data in the same data
+    structure, it becomes easier for programmers to mistakenly trust unvalidated
+    data.'
+  category: 'sast'
+  cwe: "CWE-501: Trust Boundary Violation"
+scope:
+  languages:
+    - java
+definition:
+  - and:
```
Comment (arielkru) [inline checkov/sast/checks/java/TrustBoundary.yaml:18]: redundant

File: checkov/sast/checks/java/BrokenCryptographicAlgorithm.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  category: 'sast'
+  cwe: "CWE-327: Use of a Broken or Risky Cryptographic Algorithm"
+scope:
+  languages:
+    - java
+definition:
+  - and:
+    - cond_type: "pattern"
+      operator: "equals"
+      value: class $CLAZZ extends java.security.MessageDigest {
```
Comment (arielkru) [inline checkov/sast/checks/java/BrokenCryptographicAlgorithm.yaml:15]: Im surprised this multiline works without the `|` prefix. Cool

**Review by arielkru — APPROVED** [review]
🚀

**Review by achiar99 — APPROVED** [review]

**Pull request merged**